### PR TITLE
Perform initialization on BufEnter instead of in ftplugin

### PR DESCRIPTION
**Pull request by abrace**
To address issue #110 , in this branch initialization is performed in a BufEnter autocommand instead of in the ftplugin.

What do you think about this approach?

### CODE REVIEW

**Comment by Nomad145** [comment]
Hi @abrace.

I tested this branch out with my NeoVim configuration.  It did not fix the `Ctrl-]` issue I am encountering from #110.

I had the branch installed via Plug:
```
Plug 'abrace/phpcd.vim', { 'for': 'php', 'do': 'composer install', 'branch': 'abrace-100' }
```

I can provide logs or my Vim configuration if you need them.

**Comment by abrace** [comment]
Thanks for testing, @Nomad145 . As I wrote under #110 , I think we have two different problems. The changes in this branch do resolve the specific problem I'm having with switching between files in different projects.

**Comment by lvht** [comment]
As @abrace does not respond my review, I close this PR. Feel free to reopen PR if necessary. And Feel free to make your own fork. Thank you.

**Comment by abrace** [comment]
@lvht I don't see a way reopen this PR, but am commenting because I never saw your review and cannot find it. The sidebar says "No reviews" for me. Am I missing something?

File: plugin/phpcd.vim [hunk 1]
```diff
@@ -11,6 +11,7 @@ let g:phpcd_disable_modifier = 0
 autocmd BufLeave,VimLeave *.php if g:phpcd_need_update > 0 | call phpcd#UpdateIndex() | endif
 autocmd BufWritePost *.php let g:phpcd_need_update = 1
 autocmd FileType php setlocal omnifunc=phpcd#CompletePHP
+autocmd FileType php autocmd BufEnter <buffer> call phpcd#EnterBuffer()
```
Comment (lvht) [inline plugin/phpcd.vim]: Sorry for misunderstanding.

I once commented that 
> The phpcd#EnterBuffer should only be called when the g:phpcd_auto_restart is set to 1

**Comment by lvht** [comment]
> I don't see a way reopen this PR

@abrace Feel free. This is what open source meaning.

**Pull request closed without merging**